src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -83,7 +83,7 @@ public void postTenant(TenantAttributes ta, Map<String, String> headers,
     log.info("postTenant");
     Vertx vertx = cntxt.owner();
     super.postTenant(ta, headers, res -> {
-      if (res.failed()) {
+      if (res.failed() || res.result().getStatus() >= 300) {
```

Comment:
This check will also return any status codes over 300, meaning redirection statuses will be considered failures and handled as such. Is that the intended behaviour?

Reply:
Yes.
A redirection means that a different API needs to be called to start the installation/upgrade. As no tables have been installed/upgraded no reference/sample data should be inserted.

Reply:
I think the description of this pull request and JIRA issue could be changed to explain that.

> A redirection means that a different API needs to be called to start the installation/upgrade. 

Is there an existing example of this? Should Okapi follow that redirection.

> As no tables have been installed/upgraded no reference/sample data should be inserted.

Agreed, the upgrade should fail fast.